### PR TITLE
36 allow helix index changing in batch

### PR DESCRIPTION
**Pull request by vbKenobi**
## Description

Added a view component called adjust helix indices under the group dropdown, that allows user to change helix indices. 

## Related Issue
#36-allow_helix-index-changing-in-batch

## Motivation and Context
Allows user to change the group indices under the group --> adjust helix indices menu dropdown option. 

## How Has This Been Tested?
Manually used the menu option to make sure that helix indices would change.

### CODE REVIEW

**Comment by dave-doty** [comment]
Can you fill in a bit under each heading of the PR?

**Comment by vbKenobi** [comment]
Should be updated!

**Review by dave-doty — CHANGES_REQUESTED** [review]

File: lib/src/middleware/helix_idxs_change.dart [hunk 1]
```diff
@@ -12,6 +12,24 @@ helix_idxs_change_middleware(Store<AppState> store, dynamic action, NextDispatch
...
+    var new_indices = action.idx_replacements.values.toSet();
+
+    if (new_indices.length != action.idx_replacements.length) {
+      Set<int> intsSoFar = {};
+      List<int> all_helices = action.idx_replacements.values.toList();
+
+      for (var old_index in old_idxs) {
+        var new_index = action.idx_replacements[old_index];
+        if (intsSoFar.contains(new_index)) {
+          var msg = 'First duplicate found for helix ${old_index} with new index ${new_index}. ';
```
Comment (dave-doty) [inline lib/src/middleware/helix_idxs_change.dart:24]: Sorry, try it one more time:

image

Imagine you're a user and there's a HUGE list of indices there. If they tried to assign index a to b and also index c to b, then they'll want to know all three of a, b, c. The current error message is only telling them two of these.

Try this error message: "You tried to assign helices 4 and 5 both to 0. Each helix must have a *unique* new index; make sure all the integers you write are distinct from each other and do not appear elsewhere in the design."

**Review by dave-doty — CHANGES_REQUESTED** [review]

**Review by dave-doty — CHANGES_REQUESTED** [review]
Something is still not right. I try this invalid change:

image

It gives this error message:

image

It should instead say "*You tried assigning existing helices 0 and 1 to 0*" In other words it should indicate both "source" indices that are in conflict. Otherwise the user cannot locate the conflict and fix it if they only know one of the helices causing the problem.

**Review by dave-doty — APPROVED** [review]

**Pull request merged**